packages/backend-defaults/src/entrypoints/urlReader/lib/GithubUrlReader.ts

```diff
@@ -187,6 +191,34 @@ export class GithubUrlReader implements UrlReaderService {
     }
 
     const { filepath } = parseGitUrl(url);
+
+    // If it's a direct URL we use readUrl instead
+    if (!filepath?.match(/[*?]/)) {
```

Comment:
We'll want do this check and fork before checking repo details imo, otherwise we'll end up with a lot of extra repo details requests. Along with that we'll also want to use the `etag` that's returned from the `readUrl` call instead of the commit sha. I'm not too sure what to do if the `etag` is missing though, it's a bit unfortunate that the `search` method requires it. We could try throwing an error, although I have a feeling that returning `''` is gonna be the simplest and most compatible solution.